### PR TITLE
[Topics] There are default values when user try to edit topic 

### PR DESCRIPTION
**Pull request by ssiradeghyan**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

When user try to edit topic after creation there are defaults values in edit topic settings view

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/Edit.tsx [hunk 1]
```diff
@@ -53,28 +53,6 @@ export const DEFAULTS = {
...
-      ?.filter(
-        (el) =>
-          el.value !== el.defaultValue &&
-          Object.keys(TOPIC_CUSTOM_PARAMS).includes(el.name)
-      )
-      .map((el) => ({ name: el.name, value: el.value })),
-  };
-};
-
 let formInit = false;
```
Comment (simonyandev) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/Edit.tsx:56]: I'm not sure why this is here?  I can see you reassign it `formInit = false;` on line 79 inside useEffect. This is not a good way to do, instead use useState or if you don't need to trigger rerenders, use useRef to store this variable.

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,598 @@
+import topicParamsTransformer from 'components/Topics/Topic/Edit/topicParamsTransformer';
+import { TopicWithDetailedInfo } from 'redux/interfaces';
+import { DEFAULTS } from 'components/Topics/Topic/Edit/Edit';
+import { CleanUpPolicy, ConfigSource } from 'generated-sources';
+
+const startParams: TopicWithDetailedInfo | undefined = {
```
Comment (simonyandev) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:6]: Just create a ./fixtures file somewhere near this test file to separate such huge objects from the tests.

Comment (workshur) [reply kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:6]:
```suggestion
const startParams: TopicWithDetailedInfo = {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+  config: [
+    {
+      name: 'compression.type',
+      value: 'producer',
+      defaultValue: 'producer',
+      source: 'DEFAULT_CONFIG' as ConfigSource,
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:12]:
```suggestion
      source: ConfigSource.DEFAULT_CONFIG,
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts (hunk 2 above)
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:6]: check this file. We already have fixtures
```
src/components/Topics/Topic/Edit/__test__/fixtures.ts
```

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+
+describe('topicParamsTransformer', () => {
+  it('topic not found', () => {
+    expect(topicParamsTransformer(undefined)).toEqual(DEFAULTS);
+  });
+
+  it('topic  found', () => {
+    expect(topicParamsTransformer(startParams)).toEqual(completedParams);
+  });
+});
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:598]: pls cover all the cases it is not complex

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/Edit.tsx [hunk 5]
```diff
@@ -137,7 +115,9 @@ const Edit: React.FC<Props> = ({
           <FormProvider {...methods}>
             <TopicForm
               topicName={topicName}
+              retentionBytes={defaultValues.retentionBytes}
               isSubmitting={isSubmitting}
+              cleanUpPolicy={topic.cleanUpPolicy}
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/Edit.tsx:120]: pls remove it. Props must be a part of initial values of form

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
+import topicParamsTransformer from 'components/Topics/Topic/Edit/topicParamsTransformer';
+import { DEFAULTS } from 'components/Topics/Topic/Edit/Edit';
+
+import { completedParams, topicWithInfo } from './fixtures';
+
+describe('topicParamsTransformer', () => {
+  it('topic not found', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:7]: it returns default values when topic is not defined

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    );
+    expect(
+      typeof topicParamsTransformer({
+        ...topicWithInfo,
+        partitionCount: undefined,
+      }).partitions
+    ).toEqual('number');
+  });
+
+  it('topic  maxMessageBytes', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:26]: describe maxMessageBytes
 - it 
 - it convert to number
 - it returns default value when

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      ) || -1,
+
+    [TOPIC_CUSTOM_PARAMS_PREFIX]: topic.config
+      ?.filter(
+        (el) =>
+          el.value !== el.defaultValue &&
+          Object.keys(TOPIC_CUSTOM_PARAMS).includes(el.name)
+      )
+      .map((el) => ({ name: el.name, value: el.value })),
+  };
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts:42]: test case

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+} from 'components/Topics/Topic/Edit/topicParamsTransformer';
+import { DEFAULTS } from 'components/Topics/Topic/Edit/Edit';
+import { TOPIC_CUSTOM_PARAMS_PREFIX } from 'lib/constants';
+import { ConfigSource } from 'generated-sources';
+
+import { completedParams, topicWithInfo } from './fixtures';
+
+describe('topicParamsTransformer', () => {
+  const testField = (name: keyof typeof DEFAULTS, fieldName: string) => {
+    it('return completed values', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:12]:
```suggestion
    it('returns completed values', () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+  return {
+    ...DEFAULTS,
+    name,
+    replicationFactor,
+    partitions: topic.partitionCount || DEFAULTS.partitions,
+    maxMessageBytes: getValue(topic, 'max.message.bytes') || 1000012,
+    minInsyncReplicas: getValue(topic, 'min.insync.replicas') || 1,
+    retentionBytes: getValue(topic, 'retention.bytes') || -1,
+    retentionMs: getValue(topic, 'retention.ms') || -1,
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts:23]: I would prefer to move `|| -1` logic inside of `getValue`. So add one more `defaultValue` param

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 11]
```diff
@@ -0,0 +1,99 @@
+import topicParamsTransformer, {
+  getValue,
+} from 'components/Topics/Topic/Edit/topicParamsTransformer';
+import { DEFAULTS } from 'components/Topics/Topic/Edit/Edit';
+
+import { transformedParams, customConfigs, topicWithInfo } from './fixtures';
+
+describe('topicParamsTransformer', () => {
+  const testField = (name: keyof typeof DEFAULTS, fieldName: string) => {
+    it('return transformed value', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:10]:
```suggestion
    it('returns transformed value', () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+      expect(topicParamsTransformer(topicWithInfo)[name]).toEqual(
+        transformedParams[name]
+      );
+    });
+    it(`return default value when ${name} not defined`, () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:15]:
```suggestion
    it(`returns default value when ${name} not defined`, () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+
+    it('typeof return value is number', () => {
+      expect(
+        typeof topicParamsTransformer(topicWithInfo).retentionBytes
+      ).toEqual('number');
+    });
+  };
+
+  describe('getValue', () => {
+    it('return value when find field name', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:34]:
```suggestion
    it('returns value when field exists', () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+      expect(
+        getValue(topicWithInfo, 'confluent.tier.segment.hotset.roll.min.bytes')
+      ).toEqual(104857600);
+    });
+    it('return value when not defined field name', () => {
+      expect(getValue(topicWithInfo, 'some.unsupported.fieldName')).toEqual(-1);
+    });
+    it('return value when not defined field name and has a default value', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:42]:
```suggestion
    it('returns default value when field does not exist', () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+  describe('minInsyncReplicas', () =>
+    testField('minInsyncReplicas', 'min.insync.replicas'));
+
+  describe('retentionBytes', () =>
+    testField('retentionBytes', 'retention.bytes'));
+
+  describe('retentionMs', () => testField('retentionMs', 'retention.ms'));
+
+  describe(`customParams`, () => {
+    it('return value when configs is empty', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:84]:
```suggestion
  describe('customParams', () => {
    it('returns value when configs is empty', () => {
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts [hunk 16]
```diff
@@ -0,0 +1,39 @@
+import { TopicWithDetailedInfo } from 'redux/interfaces';
+import { TOPIC_CUSTOM_PARAMS, TOPIC_CUSTOM_PARAMS_PREFIX } from 'lib/constants';
+import { DEFAULTS } from 'components/Topics/Topic/Edit/Edit';
+
+export const getValue = (
+  topic: TopicWithDetailedInfo,
+  fieldName: string,
+  defaultValue = -1
```
[the remark this answers is not in the document]
Comment (Haarolean) [reply kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts:8]: Uuhm is that like for all the int params? Seems dangerous. How did it work before?

[the remark this answers is not in the document]
Comment (ssiradeghyan) [reply kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts:8]: I added that. There was no such thing.

[the remark this answers is not in the document]
Comment (workshur) [reply kafka-ui-react-app/src/components/Topics/Topic/Edit/topicParamsTransformer.ts:8]:
```suggestion
  defaultValue: string | number
```

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+  describe('retentionMs', () => testField('retentionMs', 'retention.ms'));
+
+  describe('customParams', () => {
+    it('returns value when configs is empty', () => {
+      expect(
+        topicParamsTransformer({ ...topicWithInfo, config: [] }).customParams
+      ).toEqual([]);
+    });
+
+    it('return value when had a 2 custom configs', () => {
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/topicParamsTransformer.spec.ts:92]: returns

**Review by rAzizbekyan — APPROVED** [review]

**Comment by armenuikafka** [comment]
Tested on http://pr2205.internal.kafka-ui.provectus.io/ branch, please merge.

**Review by armenuikafka — APPROVED** [review]

**Pull request merged**